src/rpc/methods/eth.rs

```diff
@@ -1373,6 +1283,70 @@ impl RpcMethod<0> for EthSyncing {
     }
 }
 
+pub enum EthGetCode {}
+impl RpcMethod<2> for EthGetCode {
+    const NAME: &'static str = "Filecoin.EthGetCode";
+    const PARAM_NAMES: [&'static str; 2] = ["eth_address", "block_number_or_hash"];
+    const API_VERSION: ApiVersion = ApiVersion::V1;
+    const PERMISSION: Permission = Permission::Read;
+
+    type Params = (EthAddress, BlockNumberOrHash);
+    type Ok = Bytes;
+
+    async fn handle(
+        ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
+        (eth_address, block_number_or_hash): Self::Params,
+    ) -> Result<Self::Ok, ServerError> {
+        let ts = tipset_by_block_number_or_hash(&ctx.chain_store, block_number_or_hash)?;
+        let to_address = FilecoinAddress::try_from(&eth_address)?;
+        let actor = ctx
+            .state_manager
+            .get_required_actor(&to_address, *ts.parent_state())?;
+        // Not a contract. We could try to distinguish between accounts and "native" contracts here,
+        // but it's not worth it.
+        if !fil_actor_interface::is_evm_actor(&actor.code) {
```

Comment:
nit: return early here, then you can get of the huge else block that follows.

Reply:
Fixed